hackathon_site/dashboard/frontend/src/components/general/SideSheetR/SideSheetR.js

```diff
@@ -0,0 +1,165 @@
+                </FormControl>
+                <Button
+                    variant="contained"
+                    className={styles.cartButton}
+                    onClick={cart}
+                    disableElevation
+                >
+                    ADD TO CART
+                </Button>
+            </div>
+        );
+    } else {
+        return null;
+    }
+};
+
+const BodySheet = ({ detail }) => {
+    return (
+        <div className={styles.bodysheet}>
+            <div className={styles.bodyinfo}>
+                <Typography variant="h3">Manufacturer</Typography>
+                <Typography>{detail.manufacturer}</Typography>
+            </div>
+            <div className={styles.bodyinfo}>
+                <Typography variant="h3">Model Number</Typography>
+                <Typography>{detail.model_num}</Typography>
+            </div>
+            <div className={styles.bodyinfo}>
+                <Typography variant="h3">Datasheet</Typography>
+                <Link
+                    href={detail.datasheet}
+                    rel="noopener"
+                    color="inherit"
+                    underline="none"
+                    target="_blank"
+                    style={{ display: "flex" }}
+                >
+                    <LaunchIcon></LaunchIcon>
+                    <Typography>Link</Typography>
+                </Link>
+            </div>
+            <div className={styles.bodyinfo}>
+                <Typography variant="h3">Notes</Typography>
+                {detail.notes.map((note, i) => (
+                    <Typography key={i}>{note}</Typography>
+                ))}
+            </div>
+            <div className={styles.bodyinfo}>
+                <Typography variant="h3">Constraints</Typography>
+                {detail.constraints.map((constraint, i) => (
+                    <Typography key={i}>{constraint}</Typography>
+                ))}
+            </div>
+        </div>
+    );
+};
+
+const HeaderSheet = ({ detail }) => {
+    return (
+        <div className={styles.headersheet}>
+            <div>
+                <div className={styles.title}>
+                    <CircleType
+                        style={{ color: detail.type, margin: "0 5px" }}
+                    ></CircleType>
+                    <Typography variant="h2">{detail.name}</Typography>
+                </div>
+                <Typography className={styles.available}>
+                    {detail.available} OF {detail.total} IN STOCK
+                </Typography>
+                <div className={styles.tags}>
+                    <Typography variant="h3">Tags</Typography>
+                    <Typography>{detail.tags}</Typography>
+                </div>
+            </div>
+            <img src={detail.img} alt="product" />
+        </div>
+    );
+};
+
+const SideSheetR = ({ detail, cart }) => {
+    const [toggle, setToggle] = React.useState(false);
+
+    const handleChange = (change) => {
```

Comment:
This may just be a placeholder, but give functions like this a better name, for example `toggleCardState`. Though this function is not really necessary, you could just do inline:
```
onClick={() => {setToggle(false)}}
```